rtsp.go

```diff
@@ -619,19 +564,26 @@ func (rc *rtspCamera) initH264(session *description.Session) (err error) {
 				}
 			}
 		}
+	}
 
-		onPacketRTP = func(pkt *rtp.Packet) {
-			publishToWebRTC(pkt)
-			au, err := rtpDec.Decode(pkt)
-			if err != nil {
-				if !errors.Is(err, rtph264.ErrNonStartingPacketAndNoPrevious) && !errors.Is(err, rtph264.ErrMorePacketsNeeded) {
-					rc.logger.Debugf("error decoding(1) h264 rstp stream %w", err)
-				}
-				return
+	onPacketRTP := func(pkt *rtp.Packet) {
+		pts, ok := rc.client.PacketPTS2(media, pkt)
```

Comment:
I slightly changed the structure to first try to get the PTS (which is needed for all other operations) and then pass the result to the functions that care about it.